### PR TITLE
docs: Improve `@fluidframework/test-client-utils` API docs

### PR DESCRIPTION
**Pull request by Josmithr**
# Improve `@fluidframework/test-client-utils` API docs

The intention behind this PR is to improve API documentation on the consumer-facing APIs of the `@fluidframework/test-client-utils` package. It also includes some changes in transitively exposed APIs from packages upon which it depends. 

See here for the public-facing, generated docs: https://fluidframework.com/docs/apis/test-client-utils/

The TODOs and comments I left are in search of suggestions from area experts.  If you are reviewing this and know of someone who would have expertise and/or opinions on the docs, please feel free to tag them :)

[ADO #643](https://dev.azure.com/fluidframework/internal/_workitems/edit/643)

### CODE REVIEW

File: common/lib/protocol-definitions/src/users.ts [hunk 1]
```diff
@@ -7,5 +7,8 @@
  * Base user definition. It is valid to extend this interface when adding new details to the user object.
  */
 export interface IUser {
+    /**
+     * TODO
```
Comment (skylerjokiel) [inline common/lib/protocol-definitions/src/users.ts:11]:
```suggestion
     * Unique identifier of the user session. This id is established on each connection with the service.
```

**Review by skylerjokiel — APPROVED** [review]
🚀

File: api-report/test-runtime-utils.api.md [hunk 2]
```diff
@@ -67,7 +67,9 @@ export interface IMockContainerRuntimePendingMessage {
 
 // @public
 export class InsecureTokenProvider implements ITokenProvider {
-    constructor(tenantKey: string, user: IUser);
+    constructor(
+    tenantKey: string,
+    user: IUser);
```
Comment (tylerbutler) [inline api-report/test-runtime-utils.api.md:72]: Do you know how api-extractor formats its output? I've noticed these sorts of changes in other PRs, and I think they imply that we're not configuring api-extractor consistently in our configs.

File: common/lib/protocol-definitions/src/users.ts [hunk 3]
```diff
@@ -7,5 +7,8 @@
  * Base user definition. It is valid to extend this interface when adding new details to the user object.
  */
 export interface IUser {
+    /**
+     * Unique identifier of the user session. This id is established on each connection with the service.
```
Comment (tylerbutler) [inline common/lib/protocol-definitions/src/users.ts:11]:
```suggestion
     * Unique identifier of the user session. This ID is established on each connection with the service.
```

File: packages/drivers/routerlicious-driver/src/tokens.ts [hunk 4]
```diff
@@ -21,8 +21,7 @@ export interface ITokenResponse {
 }
 
 /**
- * The ITokenProvider abstracts the token fetching mechanism for a host. Host will be responsible for
- * implementing the interfaces.
+ * Abstracts the token fetching mechanism for a host. Host will be responsible for implementing the interfaces.
```
Comment (tylerbutler) [inline packages/drivers/routerlicious-driver/src/tokens.ts:24]: What is the "host" in this context?

File: packages/runtime/test-runtime-utils/src/insecureTokenProvider.ts [hunk 5]
```diff
@@ -8,17 +8,28 @@ import { ITokenProvider, ITokenResponse } from "@fluidframework/routerlicious-dr
 import { generateToken } from "./generateToken";
 
 /**
- * As the name implies this is not secure and should not be used in production. It simply makes the example easier
- * to get up and running.
+ * {@link @fluidframework/routerlicious-driver#ITokenProvider} intended for **test use only**.
+ * As the name implies this is not secure and should not be used in production.
+ * It simply makes the example easier to get up and running.
```
Comment (tylerbutler) [inline packages/runtime/test-runtime-utils/src/insecureTokenProvider.ts:13]:
```suggestion
 * It simply makes examples where authentication is not relevant easier to bootstrap.
```

File: packages/runtime/test-runtime-utils/src/insecureTokenProvider.ts [hunk 6]
```diff
@@ -8,17 +8,28 @@ import { ITokenProvider, ITokenResponse } from "@fluidframework/routerlicious-dr
...
+        /**
+         * User with whom generated tokens will be associated.
+         */
         private readonly user: IUser,
     ) {
 
     }
 
+    /**
+     * {@inheritDoc @fluidframework/routerlicious-driver#ITokenProvider.fetchOrdererToken}
```
[the remark this answers is not in the document]
Comment (ssimic2) [reply packages/runtime/test-runtime-utils/src/insecureTokenProvider.ts:31]: Are we exporting API from @fluidframework/routerlicious-driver ? Would this be a dead link?

File: packages/drivers/routerlicious-driver/src/tokens.ts (hunk 4 above)
Comment (skylerjokiel) [reply packages/drivers/routerlicious-driver/src/tokens.ts:24]: Let's clarify hosting application

```suggestion
 * Abstracts the token fetching mechanism for a hosting application. The hosting application will be responsible for providing an implementation.
```

**Review by skylerjokiel — APPROVED** [review]
1 more recommendation

File: packages/runtime/test-runtime-utils/src/insecureTokenProvider.ts [hunk 7]
```diff
@@ -8,17 +8,28 @@ import { ITokenProvider, ITokenResponse } from "@fluidframework/routerlicious-dr
...
- * to get up and running.
+ * {@link @fluidframework/routerlicious-driver#ITokenProvider} intended for **test use only**.
+ * As the name implies this is not secure and should not be used in production.
+ * It simply makes examples where authentication is not relevant easier to bootstrap.
  */
 export class InsecureTokenProvider implements ITokenProvider {
     constructor(
+        /**
+         * TODO
+         */
```
Comment (skylerjokiel) [inline packages/runtime/test-runtime-utils/src/insecureTokenProvider.ts:19]:
```suggestion
        /**
         * Private server tenantKey for generating tokens.
         */
```

**Review by skylerjokiel — APPROVED** [review]

File: packages/runtime/test-runtime-utils/src/insecureTokenProvider.ts [hunk 8]
```diff
@@ -8,17 +8,28 @@ import { ITokenProvider, ITokenResponse } from "@fluidframework/routerlicious-dr
 import { generateToken } from "./generateToken";
 
 /**
- * As the name implies this is not secure and should not be used in production. It simply makes the example easier
- * to get up and running.
+ * {@link @fluidframework/routerlicious-driver#ITokenProvider} intended for **test use only**.
```
[the remark this answers is not in the document]
Comment (skylerjokiel) [reply packages/runtime/test-runtime-utils/src/insecureTokenProvider.ts:11]:
```suggestion
 * {@link @fluidframework/routerlicious-driver#ITokenProvider} intended for **test use only**.
 * Provides an in memory implementation of the ITokenProvider interface that can be used to insecurely connect to Azure Fluid Relay (and other routerlicious based fluid relays).
```

File: packages/runtime/test-runtime-utils/src/insecureTokenProvider.ts [hunk 9]
```diff
@@ -8,17 +8,29 @@ import { ITokenProvider, ITokenResponse } from "@fluidframework/routerlicious-dr
...
  */
 export class InsecureTokenProvider implements ITokenProvider {
     constructor(
+        /**
+         * TODO
+         */
         private readonly tenantKey: string,
+
+        /**
+         * User with whom generated tokens will be associated.
```
[the remark this answers is not in the document]
Comment (skylerjokiel) [reply packages/runtime/test-runtime-utils/src/insecureTokenProvider.ts:23]: Seem good to me.

**Review by skylerjokiel — APPROVED** [review]

**Pull request merged**